homeassistant/helpers/template.py

```diff
@@ -863,6 +864,70 @@ def __repr__(self) -> str:
         return "<template AllStates>"
 
 
+class StateTranslated:
+    """Class to represent a translated state in a template."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize all states."""
+        self._hass = hass
+
+    def __call__(self, entity_id: str) -> str | None:
+        """Retrieve translated state if available."""
+        language = self._hass.config.language
+        state = _get_state_if_valid(self._hass, entity_id)
+
+        if state is None:
+            return STATE_UNKNOWN
+
+        if state.state in [STATE_UNAVAILABLE, STATE_UNKNOWN]:
+            return state.state
+
+        domain = state.domain
+        device_class = "_"
+        if "device_class" in state.attributes:
+            device_class = state.attributes["device_class"]
+
+        translations_entity_component = get_cached_translations(
+            self._hass, language, "entity_component"
+        )
```

Comment:
We only need translations for `domain`, shouldn't we just ask for those?
```suggestion
        translations_entity_component = get_cached_translations(
            self._hass, language, "entity_component", (domain,)
        )
```